File: server/Models/Task.cs
using System;
using System.Collections.Generic;

namespace server.Models;

public partial class Task
{
    public int TaskId { get; set; }

    public int UserId { get; set; }

    public string Title { get; set; } = null!;

    public string? Description { get; set; }

    public string? Status { get; set; }

    public string? Priority { get; set; }

    public DateTime? DueDate { get; set; }

    public DateTime? CreatedAt { get; set; }

    public virtual User User { get; set; } = null!;
}

File: client/src/Models/Task.ts
enum Status {
    Todo = 'Todo',
    InProgress = 'In progress',
    Done = 'Done',
}

enum Priority {
    Low = 'Low',
    Normal = 'Normal',
    High = 'High',
}

export interface Task {
    taskId: number;
    userId: number;
    title: string;
    description: string;
    status: Status;
    priority: Priority;
    dueDate: Date;
    createAt: Date;
}
